test/io/file/basename.wren

```diff
@@ -0,0 +1,61 @@
+import "io" for File
+import "os" for Platform
+
+var tests = []
+
+tests.add({"input": "", "expected": ""})  // 0
+tests.add({"input": ".", "expected": "."})  // 1
+tests.add({"input": "..", "expected": ".."})  // 2
+tests.add({"input": "file.txt", "expected": "file.txt"})  // 3
+tests.add({"input": "/", "expected": "/"})  // 4
+tests.add({"input": "/foo", "expected": "foo"}) // 5
+tests.add({"input": "/foo/", "expected": "foo"})  // 6
+tests.add({"input": "/foo/bar", "expected": "bar"}) // 7
+tests.add({"input": "/foo/bar/", "expected": "bar"})  // 8
+tests.add({"input": "/foo/bar/baz", "expected": "baz"}) // 9
+tests.add({"input": "dir1/dir2/file", "expected": "file"})  // 10
+tests.add({"input": "dir1/file"     , "expected": "file"})  // 11
+tests.add({"input": "dir1/"  , "expected": "dir1"}) // 12
+tests.add({"input": "dir1///", "expected": "dir1"}) // 13
+tests.add({"input": "/////////"  , "expected": "/"})  // 14
+tests.add({"input": "///foo"     , "expected": "foo"})  // 15
+tests.add({"input": "///foo//"   , "expected": "foo"})  // 16
+tests.add({"input": "///foo//bar", "expected": "bar"})  // 17
+
+// 2 argument signature
+tests.add({"input": "dir1/file.txt", "suffixes": [".txt"]       , "expected": "file"})  // 18
+tests.add({"input": "dir1/file.txt", "suffixes": [".c", ".txt"] , "expected": "file"})  // 19
+tests.add({"input": "dir1/file.txt", "suffixes": [".c", ".wren"], "expected": "file.txt"})  // 20
+
+
+if (Platform.isWindows) {
+  tests = tests.map {|t|
+    t["input"]    = t["input"].replace("/", "\\\\")       // ugh
+    t["expected"] = t["expected"].replace("/", "\\")
+    return t
+  }.toList
+}
+
+// tediously, we need to spell out the tests one-by-one
```

Comment:
Well, technically with `wren-console` you could have just written this with Testie.  But if we ever hope to upstream it to `Wren CLI` (which seems possible in this case) then yeah you're stuck with this.  It's super annoying.